snuba/query/parser/__init__.py

```diff
@@ -233,6 +238,7 @@ def build_selected_expressions(
         groupby=[g.expression for g in groupby_clause],
         having=having_expr,
         order_by=orderby_exprs,
+        entity_name=get_entity_key(entity).value,
```

Comment:
ok I hate the current snuba language.
So you have to do this because we need the entity schema for the legacy query parser but you need the parsed query to pick the entity.
We really need to make this parser independent to the entity schema.

For now, please keep this field empty at this point. Empty is better than filled with the wrong name.

Reply:
Yeah, requiring the entity in order to parse a query isn't great - for now it means dataset must have a default entity and we always parse the query with the default entity's data model, but switch it out later when another entity is actually the one being selected.